Eatery/Controllers/BRB/BRBViewController.swift

```diff
@@ -7,34 +7,53 @@
 //
 
 import UIKit
+import NVActivityIndicatorView
 
 class BRBViewController: UIViewController {
     
     private enum State {
         case login
         case account(BRBAccount)
     }
-    
-    private var connectionHandler: BRBConnectionHandler!
-    private var requestStart: Date?
-    
+
+    private var loggedIn = false
+
+    private var accountManager = BRBAccountManager()
     private lazy var loginViewController = BRBLoginViewController()
     private var accountViewController: BRBAccountViewController?
     
     private var state: State = .login
+    
+    private var activityIndicator: NVActivityIndicatorView!
 
+    override init(nibName nibNameOrNil: String?, bundle nibBundleOrNil: Bundle?) {
+        super.init(nibName: nil, bundle: nil)
+        activityIndicator = NVActivityIndicatorView(frame: CGRect(x: 0, y: 0, width: 22, height: 22), type: .circleStrokeSpin, color: .white)
+        let activityItem = UIBarButtonItem(customView: activityIndicator!)
+        navigationItem.setLeftBarButton(activityItem, animated: true)
+
+        accountManager.delegate = self
+        accountManager.queryBRBDataWithSavedLogin()
+
+        if let account = accountManager.getCachedAccount() {
+            setState(.account(account))
+            activityIndicator.startAnimating()
+            loggedIn = true
+        }
```

Comment:
Move lines 30 to 42 into `viewDidLoad()`?

Reply:
We want to start querying when BRBViewController is initialized, not loaded. This means that even if the user stays on the eateries page, their account information will still be fetched in the background if that makes sense

Reply:
That makes sense. 

However, to make it more clean, I would say that `activityIndicator` should be a constant to avoid force unwraps.